### PR TITLE
modernize

### PR DESCRIPTION
**Pull request by jimmywarting**
I updated the readable-stream to v3 and they now target node => 6, so i thought we can as well update to more es6 syntax

- Removed inherit & xtend dependencies for native alternativ
- Use es6 class, const, let
- Updated all dependencies
- Removed deprecated buffer construction for newer method

### CODE REVIEW

File: read.js [hunk 1]
```diff
@@ -1,92 +1,82 @@
...
-  this._generateHeaderBlocks(file, opts, function(err, blocks) {
-    // if we encountered an error, emit it
-    if (err) {
-      return readStream.emit('error', err);
+const { Readable } = require('readable-stream')
+const toBuffer = require('typedarray-to-buffer')
+
+class FileReadStream extends Readable {
+  constructor (file, opts = {}) {
+    super()
```
Comment (feross) [inline read.js]: You need to pass `opts` into `super()`

Comment (feross) [reply read.js]: Fixed it for you

**Comment by DamonOehlman** [comment]
@feross thanks for doing the review and merge - completely lost sight of this one.

@jimmywarting thanks for the PR (apologies for missing it).

**Comment by feross** [comment]
@DamonOehlman What do you think about just adding `standard` to this repo to help catch errors? (Since this PR basically converted to StandardJS style)

**Comment by feross** [comment]
There were basically no code changes required, so I'll just add `standard` and we can remove it if you disagree. I'm going to publish v5.

**Comment by feross** [comment]
5.0.0 is published.

**Comment by DamonOehlman** [comment]
> What do you think about just adding standard to this repo to help catch errors? (Since this PR basically converted to StandardJS style)

SGTM, so no need to remove it.

**Pull request merged**